experimental/MatroidRealizationSpaces/src/SelfProjectingMatroids.jl

```diff
@@ -0,0 +1,447 @@
+
+julia> selfprojecting_realization_ideal(m)
+┌ Warning: This function is slow except for small matroids!
+└ @ Oscar ~/Documents/Oscar.jl/experimental/MatroidRealizationSpaces/src/SelfProjectingMatroids.jl:169
+Ideal generated by
+  x1*x2*x3 - x1*x2*x4 - x1*x3*x4 + x1*x4 + x2*x3*x4 - x2*x3
+  ```
+"""
+function selfprojecting_realization_ideal(m::Matroid;saturate::Bool = false, check::Bool = true)::Ideal
+    @warn "This function is slow except for small matroids!"
+    if check 
+      @req is_selfprojecting(m) "The given matroid is not self projecting" 
+    end
+    if !is_realizable(m,char = 0)
+      return defining_ideal(realization_space(m,char =0, QQ))
+    end
+    RS = realization_space(m,char =0,simplify = true, saturate = true); 
+    R = ambient_ring(RS); 
+    I = defining_ideal(RS);
+    n = length(matroid_groundset(m))
+    k = rank(m)
+    RR, x,l = polynomial_ring(QQ, :x=> 1:length(gens(R)), :l=>1:n; cached=false)
+    F = hom(R, RR,x) #use this to move ideals into RR
+    G = hom(RR, R, vcat(gens(R),[0 for i in 1:n])) #use this to move them back again into the ambient ring of RS #Careful! this sets all occurences of l to zero! Use only when l is eliminated !
+    L = ideal(RR, prod(l[i] for i in 1:n))
+    MRS = realization_matrix(RS)
+    M = matrix(RR,nrows(MRS),ncols(MRS),[F(MRS[i,j]) for i in 1:nrows(MRS) for j in 1:ncols(MRS)])
+    DL = diagonal_matrix(RR,[l[i] for i in 1:n])
+    V = M*DL*transpose(M)
+    IS = ideal(RR,[V[i,j] for i in 1:k for j in 1:k]) + ideal(RR,[F(gens(I)[i]) for i in 1:length(gens(I))])
+    J = stepwise_saturation(IS,l) 
+    E = eliminate(J,l) 
+    if saturate 
+      E = stepwise_saturation(E,F.(inequations(RS)))
+    end
+    if is_zero(E)
+      return ideal(ambient_ring(RS),[0]);
+    elseif is_one(E)
+      return ideal(ambient_ring(RS),[1]);
+    else
+      return ideal(R,G.(gens(E)))
+    end
+end
+
+
+#this function sorts the list of polynomials by total degree, this helps later to reduce the list of basisminors
+function sort_by_degree(polys::Vector{<:RingElem})::Vector{<:RingElem}
+    deg(p) = hasmethod(total_degree, Tuple{typeof(p)}) ? total_degree(p) : 0
+    return sort(polys, by=deg)
+end
+
+#this function computes the minors of a given matrix for the bases in a given list.
+#this list is not reduced, i.e. there are no repetitions but the polynomials are not reduced, i.e. the list might contain x+1 and y-2 and their product
+function basisminors(M::MatElem, Bases::Vector{Vector{Int}})::Vector{<:RingElem}
+    R = base_ring(M);
+    candidates = sort_by_degree([R(det(M[1:nrows(M),Bases[i]])) for i in 1:length(Bases)])
+    multiplicativeSet= nothing
+    ineqs = [R(0)]
+    if R isa MPolyQuoRing #in this case one cannot make the multiplicativeSet using the powers_of_element, so currently the simpler choice is used and there might be double entries or products of polynomials in the list - is there a better solution for this?
+      for i in 1:length(candidates)
```

Comment:
```suggestion
      for (i, candidate) in enumerate(candidates)
```
but actually seems like you don't even need i here? besides when you get the candidate?

so you could just
```suggestion
      for candidate in candidates
```
and then adjust the code accordingly